src/App.tsx

```diff
@@ -1,61 +1,139 @@
 import './App.scss';
+import { useState } from 'react';
+import { User, Todo, CombinedTodo } from './types';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+
+const getUserById = (userId: number): User | undefined => (
+  usersFromServer.find(user => user.id === userId)
+);
+
+const combinedTodos: CombinedTodo[] = todosFromServer.map((todo: Todo) => (
+  {
+    ...todo,
+    user: getUserById(todo.userId),
+  }
+));
 
 export const App = () => {
+  const [todos, setTodos] = useState<CombinedTodo[]>(combinedTodos);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [hasUserError, setHasUserError] = useState(false);
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const getNewId = () => {
+    const lastTodoId = Math.max(...todos.map(todo => todo.id));
+
+    return lastTodoId + 1;
+  };
+
+  const handleTitleInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const cleanedValue = e.target.value.replace(/[^a-zA-Zа-яА-Я0-9\s]/g, '');
+
+    setTitle(cleanedValue);
+    setHasTitleError(false);
+  };
+
+  const handleSelectedUserInput = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(+e.target.value);
+    setHasUserError(false);
+  };
+
+  const toggleErrors = () => {
+    setHasTitleError(!title);
+    setHasUserError(!selectedUser);
+  };
+
+  const addNewTodo = (todoTitle: string, userId: number) => {
+    setTodos([
+      ...todos, {
+        id: getNewId(),
+        title: todoTitle,
+        completed: false,
+        userId,
+        user: getUserById(userId),
+      }]);
+  };
```

Comment:
To get the current state of the component, you need to get it through a callback.